src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java

```diff
@@ -0,0 +1,671 @@
+		return panelFooter;
+	}
+
+	private GoodDateChooser getJCalendarFrom() {
+		if (jCalendarFrom == null) {
+			jCalendarFrom = new GoodDateChooser(LocalDate.now(), false, false);
+			jCalendarFrom.addDateChangeListener(event -> {
+				dateFrom = jCalendarFrom.getDateStartOfDay();
+				totalRows = medicalInventoryManager.getInventoryCount(InventoryType.main.toString());
+				startIndex = 0;
+				previous.setEnabled(false);
+				if (totalRows <= PAGE_SIZE) {
+					next.setEnabled(false);
+				} else {
+					next.setEnabled(true);
+				}
+				jTableInventory.setModel(new InventoryBrowsingModel(startIndex, PAGE_SIZE));
+				initializePagesCombo();
+			});
+		}
+		return jCalendarFrom;
+	}
+
+	private GoodDateChooser getJCalendarTo() {
+		if (jCalendarTo == null) {
+			jCalendarTo = new GoodDateChooser(LocalDate.now(), false, false);
+			jCalendarTo.addDateChangeListener(event -> {
+				dateTo = jCalendarTo.getDateEndOfDay();
+				totalRows = medicalInventoryManager.getInventoryCount(InventoryType.main.toString());
+				startIndex = 0;
+				previous.setEnabled(false);
+				if (totalRows <= PAGE_SIZE) {
+					next.setEnabled(false);
+				} else {
+					next.setEnabled(true);
+				}
+				jTableInventory.setModel(new InventoryBrowsingModel(startIndex, PAGE_SIZE));
+				initializePagesCombo();
+			});
+		}
+		return jCalendarTo;
+	}
+
+	private JLabel getJLabelTo() {
+		if (jLabelTo == null) {
+			jLabelTo = new JLabel();
+			jLabelTo.setHorizontalAlignment(SwingConstants.RIGHT);
+			jLabelTo.setText(MessageBundle.getMessage("angal.common.dateto.label"));
+		}
+		return jLabelTo;
+	}
+
+	private JLabel getJLabelFrom() {
+		if (jLabelFrom == null) {
+			jLabelFrom = new JLabel();
+			jLabelFrom.setHorizontalAlignment(SwingConstants.RIGHT);
+			jLabelFrom.setText(MessageBundle.getMessage("angal.common.datefrom.label"));
+		}
+		return jLabelFrom;
+	}
+
+	private JButton getNewButton() {
+		jButtonNew = new JButton(MessageBundle.getMessage("angal.common.new.btn"));
+		jButtonNew.setMnemonic(MessageBundle.getMnemonic("angal.common.new.btn.key"));
+		jButtonNew.addActionListener(actionEvent -> {
+			String draft = InventoryStatus.draft.toString();
+			String validated = InventoryStatus.validated.toString();
+			String inventoryType = InventoryType.main.toString();
+			List<MedicalInventory> draftMedicalInventories = new ArrayList<>();
+			List<MedicalInventory> validatedMedicalInventories = new ArrayList<>();
+			try {
+				draftMedicalInventories = medicalInventoryManager.getMedicalInventoryByStatusAndInventoryType(draft, inventoryType);
+				validatedMedicalInventories = medicalInventoryManager.getMedicalInventoryByStatusAndInventoryType(validated, inventoryType);
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+			}
+			if (draftMedicalInventories.isEmpty() && validatedMedicalInventories.isEmpty()) {
+				InventoryEdit inventoryEdit = new InventoryEdit();
+				InventoryEdit.addInventoryListener(InventoryBrowser.this);
```

Comment:
```suggestion
				InventoryEdit.addInventoryListener(this);
```
unnecessary qualifier in this context